### PR TITLE
Removed logger.basicConfig calls.

### LINKED ISSUE #621: GluonTS sets the root logger log-level, it should not
## Description
GluonTS is configuring loggers, specifically it is setting the log level of the root logger. This prevents the application from properly configuring its loggers.

Generally speaking, configuring loggers should be avoided in libraries and left to the application. As per [The Hitchhiker's Guide to Python](https://docs.python-guide.org/writing/logging/#logging-in-a-library):
> Because the user, not the library, should dictate what happens when a logging event occurs, one admonition bears repeating:
>
> It is strongly advised that you do not add any handlers other than NullHandler to your library’s loggers.
>
> Best practice when instantiating loggers in a library is to only create them using the \_\_name__ global variable: the logging module creates a hierarchy of loggers using dot notation, so using \_\_name__ ensures no name collisions.

In two places at least in the code, the root logger is configured:
 - component.py, [line 41](https://github.com/awslabs/gluon-ts/blob/90587baff19a764aed02ffab752f70cf927f50e5/src/gluonts/core/component.py#L41)
 - log.py, [line 22](https://github.com/awslabs/gluon-ts/blob/90587baff19a764aed02ffab752f70cf927f50e5/src/gluonts/core/log.py#L22)

## To Reproduce
```python
import logging
log = logging.getLogger()
log.setLevel('DEBUG')

print(logging.getLevelName(log.getEffectiveLevel()))

from gluonts.model.deepar import DeepAREstimator

print(logging.getLevelName(log.getEffectiveLevel()))
```
This prints:
```
DEBUG
INFO
```
The second print should output `DEBUG` since this is how it was configured on line 3.

This prevents application using gluonts to properly configure their loggers. 

## Environment
- Operating system: Ubuntu
- Python version: 3.6
- GluonTS version: 0.4.2

### PR DESCRIPTION
**Pull request by jaheba**
*Issue #, if available:* fixes #621 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jaheba** [comment]
This github CLI is not nice, it pushes the branch to the origin by default.

**Comment by lostella** [comment]
> This github CLI is not nice, it pushes the branch to the origin by default.

that's why it is beta

File: src/gluonts/core/component.py [hunk 1]
```diff
@@ -496,17 +493,17 @@ def num_gpus(refresh=False):
     return NUM_GPUS
 
 
+@functools.lru_cache()
```
Comment (jaheba) [inline src/gluonts/core/component.py]: Could this be an issue, or is `mx.context.cpu()` always the same?

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/gluon-ts/pull/635?src=pr&el=h1) Report
> Merging [#635](https://codecov.io/gh/awslabs/gluon-ts/pull/635?src=pr&el=desc) into [master](https://codecov.io/gh/awslabs/gluon-ts/commit/24af514ecb4a767791d94b57d18b6860d9281b1a&el=desc) will **decrease** coverage by `40.78%`.
> The diff coverage is `n/a`.


[Impacted file tree graph](https://codecov.io/gh/awslabs/gluon-ts/pull/635?src=pr&el=tree)

```diff
@@             Coverage Diff             @@
##           master     #635       +/-   ##
===========================================
- Coverage   83.73%   42.95%   -40.79%     
===========================================
  Files         180      178        -2     
  Lines       10281    10276        -5     
===========================================
- Hits         8609     4414     -4195     
- Misses       1672     5862     +4190     
```

| [Impacted Files](https://codecov.io/gh/awslabs/gluon-ts/pull/635?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/gluonts/model/wavenet/\_network.py](https://codecov.io/gh/awslabs/gluon-ts/pull/635/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvbW9kZWwvd2F2ZW5ldC9fbmV0d29yay5weQ==) | `14.20% <0.00%> (-84.62%)` | :arrow_down: |
| [...uonts/monkey\_patch/monkey\_patch\_take\_along\_axis.py](https://codecov.io/gh/awslabs/gluon-ts/pull/635/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvbW9ua2V5X3BhdGNoL21vbmtleV9wYXRjaF90YWtlX2Fsb25nX2F4aXMucHk=) | `16.66% <0.00%> (-83.34%)` | :arrow_down: |
| [src/gluonts/trainer/learning\_rate\_scheduler.py](https://codecov.io/gh/awslabs/gluon-ts/pull/635/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvdHJhaW5lci9sZWFybmluZ19yYXRlX3NjaGVkdWxlci5weQ==) | `17.14% <0.00%> (-82.86%)` | :arrow_down: |
| [src/gluonts/model/deepvar/\_network.py](https://codecov.io/gh/awslabs/gluon-ts/pull/635/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvbW9kZWwvZGVlcHZhci9fbmV0d29yay5weQ==) | `18.05% <0.00%> (-81.95%)` | :arrow_down: |
| [src/gluonts/transform/split.py](https://codecov.io/gh/awslabs/gluon-ts/pull/635/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvdHJhbnNmb3JtL3NwbGl0LnB5) | `16.95% <0.00%> (-81.29%)` | :arrow_down: |
| [src/gluonts/model/deepar/\_network.py](https://codecov.io/gh/awslabs/gluon-ts/pull/635/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvbW9kZWwvZGVlcGFyL19uZXR3b3JrLnB5) | `19.84% <0.00%> (-78.58%)` | :arrow_down: |
| [src/gluonts/shell/sagemaker/path.py](https://codecov.io/gh/awslabs/gluon-ts/pull/635/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvc2hlbGwvc2FnZW1ha2VyL3BhdGgucHk=) | `22.72% <0.00%> (-77.28%)` | :arrow_down: |
| [src/gluonts/model/seq2seq/\_transform.py](https://codecov.io/gh/awslabs/gluon-ts/pull/635/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvbW9kZWwvc2VxMnNlcS9fdHJhbnNmb3JtLnB5) | `19.64% <0.00%> (-76.79%)` | :arrow_down: |
| [src/gluonts/time\_feature/lag.py](https://codecov.io/gh/awslabs/gluon-ts/pull/635/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvdGltZV9mZWF0dXJlL2xhZy5weQ==) | `17.64% <0.00%> (-76.48%)` | :arrow_down: |
| [src/gluonts/model/transformer/\_network.py](https://codecov.io/gh/awslabs/gluon-ts/pull/635/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvbW9kZWwvdHJhbnNmb3JtZXIvX25ldHdvcmsucHk=) | `22.41% <0.00%> (-75.00%)` | :arrow_down: |
| ... and [106 more](https://codecov.io/gh/awslabs/gluon-ts/pull/635/diff?src=pr&el=tree-more) | |

File: src/gluonts/core/component.py (hunk 1 above)
Comment (vafl) [reply src/gluonts/core/component.py]: Yes, should be ok.
This is to avoid logging things multiple times?

Comment (jaheba) [reply src/gluonts/core/component.py]: Yes.

**Pull request merged**